packages/contracts/contracts/voting/admin-address/AdminAddress.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import {Counters} from "@openzeppelin/contracts/utils/Counters.sol";
+
+import {PluginCloneable} from "../../core/plugin/PluginCloneable.sol";
+import {IDAO} from "../../core/IDAO.sol";
+
+/// @title AdminAddress
+/// @author Aragon Association - 2022.
+/// @notice The Admin address governance plugin, to serve as a DAO's primary full controller.
+contract AdminAddress is PluginCloneable {
+    using Counters for Counters.Counter;
+
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
+    bytes4 internal constant ADMIN_ADDRESS_INTERFACE_ID =
+        this.initialize.selector ^ this.executeProposal.selector;
+
+    bytes32 public constant ADMIN_EXECUTE_PERMISSION_ID = keccak256("ADMIN_EXECUTE_PERMISSION");
+
+    /// @notice The incrimental id for proposals and executions.
+    Counters.Counter internal proposalId;
+
+    /// @notice Emitted when a proposal is created.
+    /// @param proposalId  The ID of the proposal.
+    /// @param creator  The creator of the proposal.
+    /// @param metadata The IPFS hash pointing to the proposal metadata.
+    event ProposalCreated(uint256 indexed proposalId, address indexed creator, bytes metadata);
```

Comment:
`metadataUri`?

Reply:
are we decided that for this plugin it will be metadataUri?
are we sure we don't use metadata as arbitrary bytes?